src/generate/rust.rs

```diff
@@ -1492,3 +813,540 @@ impl<Pat: Ord + Hash + RustInputPat> Rule<Pat> {
         }
     }
 }
+
+fn impl_parse_with(name: &str, rust_slice_ty: &str, out: &mut String) {
+    put!(out, "
+    impl<'a, 'i, I: ::gll::runtime::Input<Slice = ", rust_slice_ty, ">> ", name, "<'a, 'i, I> {
```

Comment:
Did this get indented more? They were intentionally set up to match up with the resulting file, indentation-wise.